Tests/tests/test_SystemAudioPlayer.cpp

```diff
@@ -0,0 +1,1356 @@
+ * 
+ * @param[in]   :  id , url
+ * @return      :  {success: true}
+ */
+TEST_F(SAPInitializedTest, SAPGetPlayerSessionId) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection,
+        _T("open"), 
+        _T("{\"audiotype\": \"pcm\",\"sourcetype\": \"websocket\",\"playmode\": \"system\" }"),
+         response
+    ));
+
+    // Find the position of "id" in the response
+    size_t idPos = response.find("\"id\"");
+
+    if (idPos != string::npos) {
+        // Find the start and end positions of the id value
+        size_t idStart = response.find(':', idPos) + 1;
+        size_t idEnd = response.find(',', idPos);
+
+        std::string idSubstring = response.substr(idStart, idEnd - idStart);
+
+        int speechId = std::stoi(idSubstring);
+
+        std::cout << "Speech ID: " << speechId << std::endl;
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("play"), _T("{\"id\": ") + std::to_string(speechId) + _T(", \"url\": \"https://www.google.com\" }"), response));
+
+        EXPECT_EQ(response, _T("{\"success\":true}"));
+        
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getPlayerSessionId"), _T("{\"url\": \"https://www.google.com\" }"), response));
+        size_t sesPos = response.find("\"sessionId\"");
+
+        if (sesPos != string::npos) {
+ 
+            size_t sesStart = response.find(':', sesPos) + 1;
+            size_t sesEnd = response.find(',', sesPos);
+
+            std::string sesSubstring = response.substr(sesStart, sesEnd - sesStart);
+
+            int sessionId = std::stoi(sesSubstring);
+
+            std::cout << "Session ID: " << sessionId << std::endl;
+            EXPECT_EQ(speechId,sessionId);
+        } else {
+            EXPECT_TRUE(false) << "Error: 'sessionid' not found in the response.";
+        }
+    } else {
+        EXPECT_TRUE(false) << "Error: 'id' not found in the response.";
+    }
+}
+/**
+ * @name  : SAPGetPlayerSessionIdInvalid
+ * @brief : GetSessionId for non-existing url
+ * 
+ * @param[in]   :  id , url
+ * @return      :  {success: true}
+ */
+TEST_F(SAPInitializedTest, SAPGetPlayerSessionIdInvalid) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));   
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getPlayerSessionId"), _T("{\"url\": \"https://www.yahoo.com\" }"), response));
+        EXPECT_EQ(response, _T("{\"sessionId\":-1,\"success\":true}")); 
+}
+/**
+ * @name  : SAPGetPlayerSessionIdInvalidUrl
+ * @brief : GetSessionId for invalid url
+ * 
+ * @param[in]   :  id , url
+ * @return      :  {success: false}
+ */
+TEST_F(SAPInitializedTest, SAPGetPlayerSessionIdInvalidUrl) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));   
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getPlayerSessionId"), _T("{\"url\": \"invalid\" }"), response));
+        EXPECT_EQ(response, _T("{\"success\":false}")); 
+}
```

Comment:
Should there be a test case, that evaluates whether two play() requests are coming with same mode?

Reply:
Multiple AudioPlayer can be created with same mode , but when system mode playback is playing and  at that same time other player also trying to play with same mode  We will return false, Here we didn't check the player functionalities like isPlaying and two players trying to play at same time